tests/sentence_transformers/test_sentence_transformers_model_export.py

```diff
@@ -285,3 +304,98 @@ def test_default_signature_assignment(model_path, basic_model):
+    sentences = [sentence, "goodbye my friends", "i am a sentence"]
+    embedding_dim = basic_model.get_sentence_embedding_dimension()
+
+    emb0 = loaded_pyfunc.predict(sentence)
+    assert emb0.shape == (1, 1)
+    assert len(emb0.iloc[0,0]) == embedding_dim
+
+    emb1 = loaded_pyfunc.predict(sentences)
+    emb2 = loaded_pyfunc.predict(pd.Series(sentences))
+    emb3 = loaded_pyfunc.predict(pd.Series(sentences).to_numpy())
+
+    for emb in [emb1, emb2, emb3]:
+        assert emb.shape == (3, 1)
+        assert len(emb.iloc[0,0]) == embedding_dim
+        assert len(emb.iloc[1,0]) == embedding_dim
+        assert len(emb.iloc[2,0]) == embedding_dim
+
+    pd.testing.assert_frame_equal(emb1, emb2)
+    pd.testing.assert_frame_equal(emb1, emb3)
+
+
+def test_spark_udf(basic_model, model_path, spark):
+    with mlflow.start_run():
+        model_info = mlflow.sentence_transformers.log_model(basic_model, "my_model")
+    loaded_model = mlflow.pyfunc.spark_udf(
+        spark, model_info.model_uri, result_type="string", # result_type=ArrayType(FloatType())
+    )
+
+    df = spark.createDataFrame([("hello MLflow",), ("bye world",)], ["text"])
+    df = df.withColumn("embedding", loaded_model("text"))
+    pdf = df.toPandas()
+
+    emb_string = pdf.loc[0, "embedding"]
+    emb = ast.literal_eval(emb_string)
+    assert len(emb) == basic_model.get_sentence_embedding_dimension()
+
+
+# test single string input and multi string input
+@pytest.mark.parametrize("input1, input2", [
+    (["hello world"], ["goodbye world!"]),
+    (["hello world", "i am mlflow"], ["goodbye world!", "i am mlflow"]),
+])
+def test_pyfunc_serve_and_score(input1, input2, basic_model):
+    with mlflow.start_run():
+        model_info = mlflow.sentence_transformers.log_model(basic_model, "my_model")
+    loaded_pyfunc = pyfunc.load_model(model_uri=model_info.model_uri)
+    local_predict = loaded_pyfunc.predict(input1)
+
+    # Check that the giving the same string to the served model results in the same result
+    inference_data = json.dumps({"inputs": input1})
+    resp = pyfunc_serve_and_score_model(
+        model_info.model_uri,
+        data=inference_data,
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON,
+        extra_args=["--env-manager", "local"],
+    )
+    serving_result = pd.DataFrame(json.loads(resp.content.decode("utf-8"))["predictions"])
+    pd.testing.assert_frame_equal(left=local_predict, right=serving_result)
+
+    # Check that the giving a different string to the served model results in a different result
+    inference_data = json.dumps({"inputs": input2})
+    resp = pyfunc_serve_and_score_model(
+        model_info.model_uri,
+        data=inference_data,
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON,
+        extra_args=["--env-manager", "local"],
+    )
+    serving_result = pd.DataFrame(json.loads(resp.content.decode("utf-8"))["predictions"])
+    assert not local_predict.equals(serving_result)
```

Comment:
Can we use `pandas.testing.assert_frame_equal` here? It tells us what's different rather than equal or not.

Reply:
Never mind.

Reply:
That function throws an exception, so we would have to catch the exception to assert that they are not equal. Would that be better than the current code?